cmd/manager/main.go

```diff
@@ -103,14 +107,30 @@ func main() {
 		os.Exit(1)
 	}
 
+	pc := permissioncheck.NewPermissionCheck(cfg)
+
+	log.Info(fmt.Sprintf("Checking %d permissions for the operator's ServiceAccount before starting...", len(operatorconfig.OperatorPermissions)))
+
+	permissionsAreValid, err := pc.CheckPermissions(operatorconfig.OperatorPermissions)
```

Comment:
Go idiom question: Does it make sense to even have the two return values for this? If this would return false, for whatever reason, an error is also returned. Thus, if `err != nil` is the same thing as access is allowed, so maybe this can be simplified.